docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,527 @@
+
+!!! Alert
+    Open discussions per working group meetings and reviews...
+
+    - Should we be validating the messages for version (V2 in this case)?  Per @lenny-intel, do we validate incoming REST requests for the particular version of the APIs?
+        - Ans (per monthly architect's meeting of 3/28/22): yes we should be validating, but this is not done in REST either.  We should implement validation for this message communications.  We should also add validation in REST communications in the future (Levski release??).  Add this to future planning meeting topics.
+    - Should API version be in the response at all?  Per @iain-anderson, would the API version be implied in that a V2 request would mean a V2 response?
+        - Ans (per monthly architect's meeting of 3/28/22): the DTO already has API version so it will be in the payload automatically.  Additionally, it was decided to add it to the message envelope.  The reason for this is that when using REST, the path of the request includes the version.  But in message communications which lacks the URL/path, it will be beneficial to also include the version in the envelope (in addition to the payload as part of the DTO) so that it is easily determined from the envelope without having to dig around in the payload.
+    - Per core WG meeting discussion of 2/24/22 - do we really need the request id (is it redundant based on already having correlation id)?
+        - Ans (per monthly architect's meeting of 3/28/22): request id and correlation id are two different things.  Request id should be returned on the response to a request (providing traceability between single request/response).  Correlation id is used to track across an entire transaction of many service request/responses (providing traceability across a "transaction" which is often across many services).  So both are needed and should be kept.
+    - If we have request id, should it be in the payload?
+        - Ans (per monthly architect's meeting of 3/28/22): This should be kept in the message envelope (not the message payaload).
+    - Per core WG meeting discussion of 2/24/22 - do we have status code?  If so should it mimic the REST/HTTP status code responses?  Do we really want to mimic HTTP in our message bus approach?  As suggested by @farshidtz, maybe we should just have an `error` boolean and then have the message indicate the error condition. 
+        - Ans (per monthly architect's meeting of 3/28/22): We will use errorCode to provide the indication of error.  The errorCode will be 0 (no error) or 1 (indicating error) as the two enums for error conditions today.  In the future, if we determine the need for additional errors, we will add more "enums" or error code numbers.  When there is an error (with errorCode set to 1), then the payload contains a message string indicating more information about the error.  When there is no error (errorCode 0) then there is no message string in the payload. 
```

Comment:
I think this should be re-visited, as we already have enums for REST for things like:

- 400 - request is in an invalid state (note, this probably needs re-wording in our swagger documentation as I think it really means invalid request)
- 404 - resource doesn't exist
- 423 - the device is locked or down
- 500 - internal error

See: https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.1.0#/default/get_device_name__name___command_

Reply:
This was covered in at least one meeting before and the thoughts were that we didn't need the complexity of all the HTTP responses.  I'll review this issue in the 4/27 TSC.

Reply:
Implementation experimentation to be done - during implementation, some options to be explored and decided on PR review.  This issue should not hold up the approval of the ADR.  Per TSC meeting of 4/27

Reply:
Added note to the ADR that this issue will be addressed/finalized during implementation.